app/javascript/components/App.js

```diff
@@ -24,18 +40,29 @@ const App = (props) => {
       .catch((error) => console.log(error))
   }
 
-  const createGym = (gym) => {
-    console.log("this is a flag", JSON.stringify(gym))
-    fetch("/gyms", {
+  const updateGym = (gym, id) => {
+    fetch(`/gyms/${id}`, {
       body: JSON.stringify(gym),
       headers: {
         "Content-Type": "application/json"
       },
-      method: "POST"
+      method: "PATCH"
     })
-      .then(response => response.json())
-      .then(payload => this.readGyms())
-      .catch(errors => console.log(errors))
+      .then((response) => response.json())
+      .then((payload) => readGyms())
+      .catch((errors) => console.log("Gym update errors:", errors))
+  }
+  const deleteGym = (id) => {
+    fetch(`/gyms/${id}`, {
+      headers: {
+        "Content-Type": "application/json"
+      },
+      method: "DELETE"
+    })
+      .then((response) => response.json())
+      .then((payload) => readGyms())
+      .catch((error) => console.log("Gym delete errors:", error))
+      .finally(() => readGyms());
```

Comment:
I don't remember, but is this .finally() method working as its intended? If its not lets remove it.